### ISSUE TITLE
Numerous sytemctl uses

### ISSUE THREAD

**Issue by LukeHandle** [iuscommunity-pkg/httpd24u#3]
There are numerous references to `sytemctl` which make this slightly more painful to run on EL6:
https://github.com/iuscommunity-pkg/httpd24u/search?utf8=%E2%9C%93&q=systemctl&type=Code

What's the usage like for this on EL7 - I'm assuming fairly low as they have default 2.4 ..?

**Comment by cgtx** [comment]
Currently we haven't packaged this for EL7.  The systemd code was just left in place in case we get a request for that, but at this point it is untested and possibly not 100% functional.  None of the systemctl commands should be used on EL6 however.  Please let us know if you found one we missed, but looking through your search results I think we covered them all.  For example, the first search result is the logrotate file. There are actually two files for this, [httpd.logrotate](https://github.com/iuscommunity-pkg/httpd24u/blob/459d1f94c1a5da4b0b87a3c89561de49f82a8c34/SOURCES/httpd.logrotate) and [httpd.logrotate-legacy](https://github.com/iuscommunity-pkg/httpd24u/blob/459d1f94c1a5da4b0b87a3c89561de49f82a8c34/SOURCES/httpd.logrotate-legacy) ([Source2](https://github.com/iuscommunity-pkg/httpd24u/blob/459d1f94c1a5da4b0b87a3c89561de49f82a8c34/SPECS/httpd24u.spec#L51) and [Source29](https://github.com/iuscommunity-pkg/httpd24u/blob/459d1f94c1a5da4b0b87a3c89561de49f82a8c34/SPECS/httpd24u.spec#L78) respectively).  The [appropriate one is installed](https://github.com/iuscommunity-pkg/httpd24u/blob/459d1f94c1a5da4b0b87a3c89561de49f82a8c34/SPECS/httpd24u.spec#L553-L561) based on the `with_systemd` macro.

**Comment by cgtx** [comment]
I dug in some more, and it looks like we did miss one.

```
# apachectl graceful
/usr/sbin/apachectl: line 116: /usr/bin/systemctl: No such file or directory
```

The fix is probably to move [patch6](https://github.com/iuscommunity-pkg/httpd24u/blob/459d1f94c1a5da4b0b87a3c89561de49f82a8c34/SPECS/httpd24u.spec#L286) inside [this conditional](https://github.com/iuscommunity-pkg/httpd24u/blob/459d1f94c1a5da4b0b87a3c89561de49f82a8c34/SPECS/httpd24u.spec#L294-L296).  I'm going to verify a few things and fix this, and try to get the fix into testing tonight.

**Comment by LukeHandle** [comment]
A week or so ago I had been looking at this on a EL6 box and had seen a failed restart due to missing systemctl.  I was not aware of the systemd detection and I just assumeded something was amiss.

**Comment by cgtx** [comment]
It was probably the graceful restart failing that you noticed.  Regular restarts worked fine.

This is fixed in build `httpd24u-2.4.12-5.ius`, which will move to the stable repos with tonight's automation.
